main/modes/games/cosCrunch/ccmgDelivery.c

```diff
@@ -0,0 +1,255 @@
+#include "ccmgDelivery.h"
+
+static void ccmgDeliveryInitMicrogame(void);
+static void ccmgDeliveryDestroyMicrogame(void);
+static void ccmgDeliveryMainLoop(int64_t elapsedUs, uint64_t timeRemainingUs, cosCrunchMicrogameState state,
+                                 buttonEvt_t buttonEvts[], uint8_t buttonEvtCount);
+static void ccmgDeliveryBackgroundDrawCallback(int16_t x, int16_t y, int16_t w, int16_t h, int16_t up, int16_t upNum);
+
+#define MINIMUM_WAIT_US        500000
+#define MAXIMUM_WAIT_US        3000000
+#define BUTTON_PRESS_WINDOW_US 700000
+#define KNOCK_DURATION_US      400000
+#define KNOCK_DELAY_US         200000
+
+#define OUTLINE_COLOR  c222
+#define FLOOR_COLOR    c431
+#define DOOR_COLOR     c432
+#define DOORKNOB_COLOR c444
+
+#define FLOOR_TOP_Y       157
+#define FLOOR_BOTTOM_Y    CC_DRAWABLE_HEIGHT
+#define FLOOR_PLANK_COUNT 18
+#define FLOOR_PLANK_WIDTH (TFT_WIDTH / FLOOR_PLANK_COUNT)
+
+#define DOOR_WIDTH       65
+#define DOOR_X           ((TFT_WIDTH - DOOR_WIDTH) / 2)
+#define DOOR_TOP_Y       60
+#define DOORKNOB_RADIUS  4
+#define DOORKNOB_HEIGHT  45
+#define DOORKNOB_X_INSET 10
+
+#define WINDOW_TOP_Y    50
+#define WINDOW_OFFSET_X 40
+#define WINDOW_WIDTH    29
+
+#define STARBURST_ROT_US_PER_DEG  8000
+#define TUMBLEWEED_X_OFFSET       ((TFT_WIDTH - DOOR_WIDTH) / 2 - 20)
+#define TUMBLEWEED_X_US_PER_PX    20000
+#define TUMBLEWEED_Y_OFFSET       (FLOOR_TOP_Y - 35)
+#define TUMBLEWEED_Y_RANGE        15
+#define TUMBLEWEED_ROT_US_PER_DEG 12000
+
+static const char ccmgDeliveryVerb[]       = "Delivery";
+static const char ccmgDeliverySuccessMsg[] = "Package Get!";
+static const char ccmgDeliveryKnock[]      = "KNOCK";
+const cosCrunchMicrogame_t ccmgDelivery    = {
+       .verb                     = ccmgDeliveryVerb,
+       .successMsg               = ccmgDeliverySuccessMsg,
+       .failureMsg               = NULL,
+       .timeoutUs                = MAXIMUM_WAIT_US + BUTTON_PRESS_WINDOW_US + 500000,
+       .fnInitMicrogame          = ccmgDeliveryInitMicrogame,
+       .fnDestroyMicrogame       = ccmgDeliveryDestroyMicrogame,
+       .fnMainLoop               = ccmgDeliveryMainLoop,
+       .fnBackgroundDrawCallback = ccmgDeliveryBackgroundDrawCallback,
+       .fnMicrogameTimeout       = NULL,
+};
+
+paletteColor_t const timeOfDayColors[] = {
+    // Daytime
+    c355,
+    // Nighttime
+    c013,
+};
+
+paletteColor_t const wallColors[] = {
+    // Pea soup yellow
+    c442,
+    // Mint green
+    c232,
+    // Simpsons' living room pink
```

Comment:
You can't see it but I have a tilted sailboat print hanging on my wall. My wife hates it, so it's been banished to the office instead of over the couch.